chia/daemon/keychain_server.py

```diff
@@ -234,6 +240,37 @@ async def add_private_key(self, request: Dict[str, Any]) -> Dict[str, Any]:
 
         return {"success": True, "fingerprint": sk.get_g1().get_fingerprint()}
 
+    async def add_public_key(self, request: Dict[str, Any]) -> Dict[str, Any]:
+        if self.get_keychain_for_request(request).is_keyring_locked():  # pragma: no cover
+            return {"success": False, "error": KEYCHAIN_ERR_LOCKED}
+
+        public_key = request.get("public_key", None)
+        label = request.get("label", None)
+
+        if public_key is None:  # pragma: no cover
+            return {
+                "success": False,
+                "error": KEYCHAIN_ERR_MALFORMED_REQUEST,
+                "error_details": {"message": "missing public_key"},
+            }
+
+        try:
+            pk = self.get_keychain_for_request(request).add_public_key(public_key, label)
+        except KeyError as e:  # pragma: no cover
+            return {
+                "success": False,
+                "error": KEYCHAIN_ERR_KEYERROR,
+                "error_details": {"message": f"The word '{e.args[0]}' is incorrect.'", "word": e.args[0]},
```

Comment:
this seems a curious error message - what `word` are you referring to here?

Reply:
Not sure but I didn't write it and (now) didn't touch it so none of my business.